misk/src/main/kotlin/misk/client/ClientInterceptor.kt

```diff
@@ -0,0 +1,18 @@
+package misk.client
+
+import okhttp3.Interceptor
+
+object ClientInterceptors {
+
+  /**
+   * This interface is used with Guice multibindings. Register instances by calling `multibind()`
+   * in a `KAbstractModule`:
+   *
+   * ```
+   * multibind<ClientInterceptors.Factory>().to<MyFactory>()
+   * ```
+   */
+  interface Factory {
```

Comment:
If we wanted to support GRPC interceptors too, could have separate factory types:

```
  interface Factory {
    fun create(action: ClientAction): Interceptor?
  }

  interface GRPCFactory {
    fun create(grpcClientClass: KClass<*>): Interceptor?
  }
```